app/controllers/users/passwords_controller.rb

```diff
@@ -2,19 +2,29 @@ module Users
   class PasswordsController < Devise::PasswordsController
     include ValidEmailParameter
 
-    before_action :confirm_valid_token, only: [:edit]
-
     def create
-      RequestPasswordReset.new(params[:user][:email]).perform
+      email = params[:user][:email]
+
+      RequestPasswordReset.new(email).perform
+
+      analytics_user = User.find_by_email(email) || AnonymousUser.new
```

Comment:
The other reason to use a Null Object is that the analytics class expects the user to have a uuid. Without the Null Object, we would need yet another conditional in the analytics class. The Null Object allows us to just send the messages and get the correct response back without cluttering the code with conditionals.